pyanaconda/kickstart_dispatcher.py

```diff
@@ -0,0 +1,359 @@
+from pykickstart.sections import Section
+
+KickstartElement = namedtuple('KickstartElement',
+                              ['args', 'content', 'lineno', 'filename'])
+
+
+class KickstartElements(object):
+    """Container for kickstart elements (commands, sections, addons)
+
+    Preserves order of added elements. Allows filtering of elements.
+    """
+
+    def __init__(self):
+        self._elements = []
+
+    def append_command(self, name, content, lineno, filename):
+        """Appends kickstart command to the container.
+
+        :param name: name of the command
+        :type name: str
+        :param content: command line including newline
+        :type content: str
+        :param lineno: line number of the command in a kickstart file
+        :type content: int
+        :param filename: name of a kickstart file
+        :type filename: str
+        """
+        command = KickstartElement(args=(name,),
+                                   content=content,
+                                   lineno=lineno,
+                                   filename=filename)
+        self._elements.append(command)
+
+    def append_section(self, name, content, header_lineno, filename):
+        """Appends kickstart section to the container.
+
+        :param name: name of the section (including leading '%' char)
+        :type name: str
+        :param content: section content (ending with newline)
+        :type content: str
+        :param header_lineno: line number of the section header in a kickstart file
+        :type content: int
+        :param filename: name of a kickstart file
+        :type filename: str
+        """
+        section = KickstartElement(args=(name,),
+                                   content=content,
+                                   lineno=header_lineno,
+                                   filename=filename)
+        self._elements.append(section)
+
+    def append_addon(self, name, content, header_lineno, filename):
+        """Appends addon kickstart section to the container.
+
+        :param name: name of the addon
+        :type name: str
+        :param content: addon section content (ending with newline)
+        :type content: str
+        :param header_lineno: line number of the addon section header in a kickstart file
+        :type content: int
+        :param filename: name of a kickstart file
+        :type filename: str
+        """
+        addon = KickstartElement(args=("%addon", name),
+                                 content=content,
+                                 lineno=header_lineno,
+                                 filename=filename)
+        self._elements.append(addon)
+
+    @property
+    def all_elements(self):
+        """List of all elements in the container.
+
+        :return: list of all elements in the container
+        :rtype: list(KickstartElement)
+        """
+
+        return list(self._elements)
+
+    def get_elements(self, commands=None, sections=None, addons=None):
```

Comment:
We probably shouldn't allow to request the `%addon` section in `sections`.

Reply:
Hmm, I agree.